2 Niagara Fundamentals Exercise - Multiplexer/multiplexer-rt/src/com/tridiumuniversity/multiplexer/BMultiplexer.java

```diff
@@ -0,0 +1,284 @@
+package com.tridiumuniversity.multiplexer;
+
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.nre.annotations.NiagaraProperty;
+import javax.baja.sys.*;
+import java.util.ArrayList;
+import java.util.logging.Logger;
+
+@NiagaraType
+@NiagaraProperty(
+        name = "switchTotal",
+        type = "BInteger",
+        defaultValue = "BInteger.DEFAULT",
+        flags = Flags.READONLY
+)
+@NiagaraProperty(
+        name = "usableSwitches",
+        type = "BInteger",
+        defaultValue = "BInteger.DEFAULT",
+        flags = Flags.READONLY
+)
+@NiagaraProperty(
+        name = "out",
+        type = "BBoolean",
+        defaultValue = "BBoolean.DEFAULT",
+        flags = Flags.SUMMARY | Flags.READONLY
+)
+public class BMultiplexer extends BComponent
+{
+    @Override
+    public void added(Property p, Context cx)
+    {
+        calcSInputCount();
+        calcInInputCount();
+        calcInputStep();
+        calcSwitchSum();
+        updateOut();
+    }
+    @Override
+    public void removed(Property p, BValue v, Context cx)
+    {
+        calcSInputCount();
+        calcInInputCount();
+        calcInputStep();
+        calcSwitchSum();
+        updateOut();
+    }
+    @Override
+    public void renamed(Property p, String n, Context cx)
+    {
+        calcSInputCount();
+        calcInInputCount();
+        calcInputStep();
+        calcSwitchSum();
+        updateOut();
+    }
+    @Override
+    public void changed(Property p, Context cx)
+    {
+        calcSwitchSum();
+        updateOut();
+    }
+
+
+    /* Calculates how many "in" properties exist in the multiplexer */
+    private void calcInInputCount()
+    {
+        inInputCount = 0; // reset the old count
+        inInputs.clear(); // clear the old "in" input list
+        SlotCursor<Property> propertyCursor = getProperties(); // create the cursor
+
+        while (propertyCursor.next()) // iterate through the properties
+        {
+            Property property = propertyCursor.property();
+            try
+            {
+                BBoolean value = (BBoolean) propertyCursor.get(); // get property value. If the property isn't a BBoolean, skip
```

Comment:
Rather than use the try/catch block to handle the case where the current Property isn't a `BBoolean`, it would probably be more elegant to do something like:

```java
if (get(property) instanceof BBoolean)
{
  /* ... */
}
```

Reply:
instanceof,
that's a useful one